File: lib/generate_units.js
const projects = [
    { name: "Samarth Heights", tower: "Tower A", start_floor: 3, end_floor: 9, units_per_floor: 4, prefix: "A", types: ["2 BHK", "3 BHK"], prices: [7500000, 10500000] },
    { name: "Samarth Heights", tower: "Tower B", start_floor: 2, end_floor: 6, units_per_floor: 4, prefix: "B", types: ["2 BHK", "4 BHK"], prices: [8100000, 16200000] },
    { name: "Samarth Plaza", tower: "Wing A (Retail)", start_floor: 1, end_floor: 3, units_per_floor: 6, prefix: "S", types: ["Shop", "Office"], prices: [12500000, 18000000] },
];

const statuses = ["available", "reserved", "booked", "sold"];
const amenities_list = ["Balcony", "Parking", "Club Access", "Garden View", "Main Road Facing"];
let uid_counter = 100;

for (const p of projects) {
    for (let floor = p.start_floor; floor <= p.end_floor; floor++) {
        for (let i = 1; i <= p.units_per_floor; i++) {
            uid_counter++;
            const u_type = p.types[Math.floor(Math.random() * p.types.length)];
            const price = p.prices[p.types.indexOf(u_type)] + (floor * 50000);
            const status = statuses[Math.floor(Math.random() * statuses.length)];
            const unit_no = `${p.prefix}-${floor}0${i}`;
            const carpet = u_type.includes("2 BHK") ? 850 : (u_type.includes("3 BHK") ? 1200 : (u_type.includes("4 BHK") ? 1800 : 500));
            const price_sqft = Math.floor(price / carpet);
            const amenities = amenities_list.slice(0, 2);

            console.log(`  {`);
            console.log(`    id: "u${uid_counter}",`);
            console.log(`    unitNo: "${unit_no}",`);
            console.log(`    projectName: "${p.name}",`);
            console.log(`    towerName: "${p.tower}",`);
            console.log(`    floor: ${floor},`);
            console.log(`    type: "${u_type}",`);
            console.log(`    carpetArea: ${carpet},`);
            console.log(`    price: ${price},`);
            console.log(`    pricePerSqFt: ${price_sqft},`);
            console.log(`    status: "${status}",`);
            console.log(`    statusHistory: [{ date: "2024-01-15", status: "Listed", user: "System" }],`);
            console.log(`    amenities: ${JSON.stringify(amenities)}`);
            console.log(`  },`);
        }
    }
}

File: lib/generate_units.py
import random

projects = [
    {"name": "Samarth Heights", "tower": "Tower A", "start_floor": 3, "end_floor": 9, "units_per_floor": 4, "prefix": "A", "types": ["2 BHK", "3 BHK"], "prices": [7500000, 10500000]},
    {"name": "Samarth Heights", "tower": "Tower B", "start_floor": 2, "end_floor": 6, "units_per_floor": 4, "prefix": "B", "types": ["2 BHK", "4 BHK"], "prices": [8100000, 16200000]},
    {"name": "Samarth Plaza", "tower": "Wing A (Retail)", "start_floor": 1, "end_floor": 3, "units_per_floor": 6, "prefix": "S", "types": ["Shop", "Office"], "prices": [12500000, 18000000]},
]

statuses = ["available", "reserved", "booked", "sold"]
amenities_list = ["Balcony", "Parking", "Club Access", "Garden View", "Main Road Facing"]
uid_counter = 100

for p in projects:
    for floor in range(p["start_floor"], p["end_floor"] + 1):
        for i in range(1, p["units_per_floor"] + 1):
            uid_counter += 1
            u_type = random.choice(p["types"])
            price = p["prices"][p["types"].index(u_type)] + (floor * 50000)
            status = random.choice(statuses)
            unit_no = f"{p['prefix']}-{floor}0{i}"
            carpet = 850 if "2 BHK" in u_type else (1200 if "3 BHK" in u_type else (1800 if "4 BHK" in u_type else 500))
            price_sqft = int(price/carpet)
            amenities = random.sample(amenities_list, 2)
            
            # Format: { prop: value }
            print(f'  {{')
            print(f'    id: "u{uid_counter}",')
            print(f'    unitNo: "{unit_no}",')
            print(f'    projectName: "{p["name"]}",')
            print(f'    towerName: "{p["tower"]}",')
            print(f'    floor: {floor},')
            print(f'    type: "{u_type}",')
            print(f'    carpetArea: {carpet},')
            print(f'    price: {price},')
            print(f'    pricePerSqFt: {price_sqft},')
            print(f'    status: "{status}",')
            print(f'    statusHistory: [{{ date: "2024-01-15", status: "Listed", user: "System" }}],')
            print(f'    amenities: {str(amenities).replace("\'", "\"")}')
            print(f'  }},')
